docs/2_5_fmu_distribution.adoc

```diff
@@ -8,7 +8,7 @@ The FMU must implement all common API functions according to <<general-mechanism
 Especially it is required that all functions that are part of the specified FMI interface type are present, even if they are only needed for optional capabilities that the FMU does not support.
 The behavior of those functions is unspecified, so while calling environments can rely on the functions being present, they cannot rely on any particular behavior for functions only needed for capabilities the FMU does not support.
 The extension of the ZIP file must be `.fmu` _[, for example, `HybridVehicle.fmu`]_.
-The compression method used for the ZIP file must be `deflate` _[(most free tools, such as zlib, offer only the common compression method `deflate`)]_.
+The compression method for all files stored in the ZIP file must be `deflated` _[(most free tools, such as zlib, offer only the common compression method `deflated`)]_.
```

Comment:
I would suggest the following wording, since very small files/directory entries will use method 0 (stored), and this specifies the other requirements, like no ZIP64, no encryption, no splitting, etc., thus achieving the broard compatibility that was intended:

```suggestion
The compression method for all files stored in the ZIP archive must be either 8 (`deflated`), or 0 (`stored`).
The version needed to extract filed of the archive must not be higher than 2.0, and encryption must not be employed.
The archive may not be a split or spanned ZIP archive.
 _[These restrictions ensure broad compatibility of the archive with common ZIP processing tools and libraries.)]_
```

Reply:
I am fine with compression and splitting, but I even thought ZIP encrpytion is legal and the simplest way to achieve some kind of FMU encryption (between trusted partners).

Reply:
The problem with allowing encryption is that all FMI implementations would then have to support encryption, which currently is definitely not what is implemented, nor what many common libraries and tools support out of the box.

Furthermore legacy ZIP encryption (which is what is in ZIP 2.0) is basically useless due to e.g. Biham and Kocher's known plaintext attack, which just requires 12 bytes of known plaintext: This is trivial for FMUs that must contain the modelDescription.xml with its XML preamble, or many other sources of known plaintext. See https://github.com/kimci86/bkcrack for working code.

And all advanced encryption support in ZIP requires newer versions of the ZIP specification, thereby more or less placing you outside the widely supported subset anyways (they employ whole directory/archive encryption concepts, which are way more secure, but not widely supported).

Nothing prevents partners/tools that want to support more than the basic spec to allow encryption (and/or more compression algos, ZIP64, etc.), but requiring this across the board seems not useful. Furthermore there are better mechanisms inside FMUs to encrypt and/or protect content in other ways, which do not require the support by the importer.

Just my two centieuros...

Reply:
Would it be of interest to update 1.0 and 2.0 to support store option?
I recently had to change our zip routine as it used store on certain files (zip files)  where the deflate was pointless. However an importer pointed out to me that that was illegal and their tool broke since their unzip routine wasn't  able to handle the split.
So if we will support this for 3.0 should we update 1.0,x and 2.0.x in the same way?

Reply:
> So if we will support this for 3.0 should we update 1.0,x and 2.0.x in the same way?

I think we should so clarify, since otherwise nearly no compliant FMUs are out there: Directory entries are usually stored as stored (sic) in nearly all zip utilities (dito for very small files, BTW), so this would make all manually built FMUs and most other FMUs illegal...